Common/src/main/java/com/tabnineCommon/chat/ChatToolWindowListener.kt

```diff
@@ -0,0 +1,45 @@
+package com.tabnineCommon.chat
+
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.wm.ToolWindow
+import com.intellij.openapi.wm.ex.ToolWindowManagerListener
+import com.tabnineCommon.capabilities.CapabilitiesService
+import com.tabnineCommon.lifecycle.BinaryStateChangeNotifier
+
+class ChatToolWindowListener : ToolWindowManagerListener {
+    companion object {
+        const val TABNINE_CHAT_TOOL_WINDOW_ID = "Tabnine Chat"
+        const val MINIMAL_MS_BETWEEN_FORCE_REFRESH_CAPABILITIES = 2_000
+    }
+
+    private var isLoggedIn = false
+    private var lastForceRefreshCapabilities = System.currentTimeMillis()
+
+    init {
+        ApplicationManager.getApplication().messageBus.connect()
+            .subscribe(
+                BinaryStateChangeNotifier.STATE_CHANGED_TOPIC,
+                BinaryStateChangeNotifier { isLoggedIn = it.isLoggedIn ?: false }
```

Comment:
where is it used ?

Reply:
In the handle tabnine toolWIndow shown, the object property maintains the current logged in status by the topic, to be used when the tabnine chat toolWindow is shown.

Reply:
cool